packages/http-client-csharp/emitter/src/type/type-cache.ts

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { SdkType } from "@azure-tools/typespec-client-generator-core";
+import { InputEnumType, InputModelType, InputType } from "./input-type.js";
+
+export interface TypeCache {
+  cache: Map<SdkType, InputType>;
+  models: Map<string, InputModelType>;
+  enums: Map<string, InputEnumType>;
+
+  has(type: SdkType): boolean;
```

Comment:
do we need this?
can it just be like `if(get(type)) {...}`

Reply:
Emmm I do not know
I copy this function from `Map<K, V>` and in your way we will have to assign a variable to it.

Reply:
I am fine to keep it.

Reply:
Actually, why do we need them at all?
since we have `types: new Map<SdkType, InputType>()` in place, can we just use `types.has(type)`, `types.get(type)` and `types.set(type, value)`? just like models and enums.

Reply:
I just think it is quite inconvenient to use initially because it is used in so many places, therefore I added them.
Since now there is only one place using it, I could remove it